prusti-viper/src/encoder/backtranslation.rs

```diff
@@ -0,0 +1,457 @@
+                            //need to find a discriminant to do something
+                            if !opt_discriminant.is_some(){
+                                //try to find disc in the associated local variable
+                                let opt_discr_locations = self.disc_info.get(&(self.def_id, vir_name.clone()));
+                                if let Some(discr_locations) = opt_discr_locations {
+                                    for name in discr_locations {
+                                        let disc_entry = silicon_ce_entries.get(name);
+                                        let value = self.backtranslate_int(disc_entry);
+                                        if value.is_some() {
+                                            opt_discriminant = value;
+                                            break;
+                                        }
+                                    }
+                                }
+                            }
+                            if let Some(x) = opt_discriminant {
+                                let discriminant = x as u32;
+                                //is this a risk? I assume discriminant will not be too large
+                                for def in &adt_def.variants{
+                                    let discr = match def.discr {
+                                        ty::VariantDiscr::Relative(d) => d,
+                                        _ => unimplemented!()  
+                                    };
+                                    if discr == discriminant {
+                                        variant = Some(def);
+                                        variant_name = def.ident.name.to_ident_string();
+                                    }
+                                }
+                            }
+                            
+                           
+
+
+                            if let Some(var_def) = variant {
+                                let sil_name = format!("enum_{}", variant_name).to_string();
+                                let opt_enum_entry = map.get(&sil_name);
+                                //at this point it should be a subroutine same for structs and enum:
+                                let result = self.backtranslate_vardef(
+                                    var_def,
+                                    opt_enum_entry,
+                                    vir_name,
+                                    subst,
+                                    silicon_ce_entries,
+                                );
+                                field_names = result.0;
+                                field_entries = result.1;
+                            }
+                        }
+                        //valid names can not start with integers (?)
+                        let named_fields = field_names.len() > 0 && !field_names[0].parse::<usize>().is_ok();
```

Comment:
This is not very nice… If we check based on the string value, we can do it when actually printing the `Enum` later, it does not need to be a field.